internal/ingress/controller/controller.go

```diff
@@ -1102,6 +1120,57 @@ func (n *NGINXController) createServers(data []*extensions.Ingress,
 	return servers
 }
 
+// match and merge virtual backends into their respective real backends
```

Comment:
can you add more description here explaining how it matches, how it merges. Ideally with examples